sphinx/domains/python.py

```diff
@@ -641,6 +641,152 @@ def generate(self, docnames=None):
         return sorted_content, collapse
 
 
+class PythonClassIndex(Index):
```

Comment:
I would rather write a function generating a class (a functor), to be able to easily have subclasses of `Index` for classes, functions, data, attributes, methods, classmethods, staticmethods etc.